src/main/java/uk/gov/hmcts/reform/et/syaapi/service/ApplicationService.java

```diff
@@ -112,15 +112,21 @@ public CaseDetails respondToApplication(String authorization, RespondToApplicati
             throw new IllegalArgumentException("Application id provided is incorrect");
         }
 
+        boolean hasIncludedFile = request.getResponse().getSupportingMaterial() != null;
+        if (!hasIncludedFile && request.getResponse().getResponse() == null) {
+            throw new IllegalArgumentException("SupportingMaterialFile or "
```

Comment:
The exception message seems a bit a clear to me. Also is it worth logging it as an error or warning ( ideally with application/case id as well)?

Reply:
actually why not just set boolean hasIncludedFile = request.getResponse().getSupportingMaterial() != null && request.getResponse().getResponse() == null; ?